### PR TITLE
feat(api-idorslug): Created IdOrSlug Lookup

### PR DESCRIPTION
**Pull request by iamrajjoshi**
The lookup will be used in `convert_args` functions to handle resource id as well as slugs in our apis
Something like:
```
doc_integration = DocIntegration.objects.get(slug__id_or_slug=doc_integration_slug)
```

For: https://github.com/getsentry/team-core-product-foundations/issues/194

### CODE REVIEW

**Review by schew2381 — APPROVED** [review]
lgtm, we should also add tests for each model as we roll them out in PRs.

Also maybe manually test this really quick to see that it actually works with a model

**Comment by iamrajjoshi** [comment]
> lgtm, we should also add tests for each model as we roll them out in PRs.
> 
> Also maybe manually test this really quick to see that it actually works with a model

Yep, I did test it manually, but wanted to keep this test model agnostic

File: src/sentry/db/models/fields/slug.py [hunk 1]
```diff
@@ -9,3 +10,22 @@ class SentrySlugField(SlugField):
...
+
+class IdOrSlugLookup(Lookup):
+    lookup_name = "id_or_slug"
+
+    def as_sql(self, compiler, connection):
+        # We can ignore the lhs because we know that the lhs will
+        # query either 'slug' or 'id'
+        rhs, rhs_params = self.process_rhs(compiler, connection)
+
+        if rhs_params and rhs_params[0].isnumeric():
```
Comment (schew2381) [inline src/sentry/db/models/fields/slug.py]: mypy error here

Comment (iamrajjoshi) [reply src/sentry/db/models/fields/slug.py]: just fixed it

**Pull request merged**